_source/_posts/2021-12-06-spring-data-elasticsearch.md

```diff
@@ -0,0 +1,349 @@
+---
+layout: blog_post
+title: "A Quick Guide to Elasticsearch for Spring Data and Spring Boot"
+author: jimena-garbarino
+by: contractor
+communities: [devops,security,mobile,.net,java,javascript,go,php,python,ruby]
+description: ""
+tags: []
+tweets:
+- ""
+- ""
+- ""
+image:
+type: conversion
+---
+
+For sure you have heard about Elasticsearch or the Elastic Stack. It seems it started as a search engine based on Lucene (an open-source search engine library) built by Shay Banon to index his wife's cooking recipes. From the early days, Elasticsearch has gone a long way and evolved into the Elastic Stack, a great suite for taking data from any source, search, analyze and visualize it in near real-time. So Elasticsearch is a distributed document store, data is serialized as JSON documents and stored distributed across cluster nodes. The _inverted index_ is a data structure that lists every unique word that appears in any document, and also the documents each word appears in. This gives support to fast full-text searches, a feature not supported or partially supported by database engines. The index is a collection of documents and each document is a collection of fields. In turn, each field can be indexed in an optimal data structure, for example, an inverted index for text fields, but a BKD tree for a numeric field.
+
+How can be Elasticsearch integrated into a Spring Boot Application? What are the options for adding Elasticsearch to a Java application? This post will give you a quick introduction to Elasticsearch integration options.
+
+**Prerequisites**:
+
+- [HTTPie](https://httpie.io/)
+- [JHipster 7](https://www.jhipster.tech/installation/)
+- [Java 14+](https://openjdk.java.net/install/index.html)
```

Comment:
Can we bump this to 17?

Reply:
> Can we bump this to 17?

I think the generator sets java 11, I have jhipster 7.5.0 and openjdk 17 in my environment and the pom.xml is generated with java 11, will double check

Reply:
You're right. Java 11 is fine.